### PR TITLE
Add projects servlet testing.

### PR DESCRIPTION
**Pull request by dtjanaka**

### CODE REVIEW

File: src/test/java/com/google/sps/servlets/BlobServletTest.java [hunk 1]
```diff
@@ -32,8 +32,8 @@
 
   private final LocalServiceTestHelper helper =
       new LocalServiceTestHelper(new LocalUserServiceTestConfig(),
-                                 new LocalDatastoreServiceTestConfig(),
-                                 new LocalBlobstoreServiceTestConfig())
+                                 new LocalDatastoreServiceTestConfig()/*,
+                                 new LocalBlobstoreServiceTestConfig()*/)
```
Comment (charmychancey) [inline src/test/java/com/google/sps/servlets/BlobServletTest.java:36]: Is this supposed to be here?

Comment (dtjanaka) [reply src/test/java/com/google/sps/servlets/BlobServletTest.java:36]: Yes, the blobstore POST testing is a WIP

**Comment by charmychancey** [comment]
lgtm

File: src/test/java/com/google/sps/servlets/ProjectServletTest.java [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+  public void delete() throws IOException, EntityNotFoundException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    assertEquals(
+        5, datastore.prepare(new Query(DataUtils.PROJECT)).countEntities());
+    when(request.getParameter("mode")).thenReturn("update");
+    when(request.getParameter("proj-id")).thenReturn(projId);
+    when(request.getParameter("delete")).thenReturn("true");
+    servlet.doPost(request, response);
+    assertEquals(
+        4, datastore.prepare(new Query(DataUtils.PROJECT)).countEntities());
```
Comment (robotsaurus) [inline src/test/java/com/google/sps/servlets/ProjectServletTest.java:154]: Can you also test requesting the specific project id and ensure it doesn't return it?

File: src/test/java/com/google/sps/ProjectServletTestUtils.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  public static final String expectedRoleEditor = "[\n  {\n    \"name\": \"MyProject3\",\n    \"timestamp\": \"2020-08-12T05:39:02.387Z\",\n    \"visibility\": \"private\",\n    \"owners\": [\n      \"xyz@xyz.com\"\n    ],\n    \"editors\": [\n      \"abc@xyz.com\"\n    ]\n  }\n]\n";
+  public static final String expectedRoleOwner = "[\n  {\n    \"name\": \"MyProject2\",\n    \"timestamp\": \"2020-08-12T05:39:02.384Z\",\n    \"visibility\": \"public\",\n    \"owners\": [\n      \"abc@xyz.com\"\n    ],\n    \"editors\": [\n      \"uvw@def.com\"\n    ]\n  },\n  {\n    \"name\": \"MyProject\",\n    \"timestamp\": \"2020-08-12T05:39:02.383Z\",\n    \"visibility\": \"private\",\n    \"owners\": [\n      \"abc@xyz.com\",\n      \"def@uvw.com\"\n    ],\n    \"editors\": [\n      \"xyz@abc.com\",\n      \"uvw@def.com\"\n    ]\n  }\n]\n";
+  public static final String expectedSearchTerm = "[\n  {\n    \"name\": \"MyProject\",\n    \"timestamp\": \"2020-08-12T05:39:02.383Z\",\n    \"visibility\": \"private\",\n    \"owners\": [\n      \"abc@xyz.com\",\n      \"def@uvw.com\"\n    ],\n    \"editors\": [\n      \"xyz@abc.com\",\n      \"uvw@def.com\"\n    ]\n  }\n]\n";
```
Comment (robotsaurus) [inline src/test/java/com/google/sps/ProjectServletTestUtils.java:20]: Is there a more maintainable way to have these expected Json strings? 

For example, can we have a java class that we then convert to json using gson?

**Pull request merged**